Freeze `k8s.gcr.io` image registry (#39214)
kubernetes/website: content/en/blog/_posts/2023-02-06-k8s-gcr-io-freeze-announcement.md

---
layout: blog
title: "k8s.gcr.io image registry will be frozen from the 3rd of April 2023"
date: 2023-02-06
slug: k8s-gcr-io-freeze-announcement
---

**Authors**: Mahamed Ali (Rackspace Technology)

> The Kubernetes project has released a new community-owned image registry called `registry.k8s.io`. This registry has been generally available(GA) for several months, and we have published a [blog post](https://kubernetes.io/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/) about it, its benefits to the community, and the Kubernetes project. This post also announced that future versions of Kubernetes will not be available in the old registry.
> 
> On the 3rd of April 2023, the `k8s.gcr.io` image registry will be frozen and no further images for Kubernetes and related subprojects will be pushed to the old registry.

Reviewer: I'd suggest to start with what happens

```suggestion
The Kubernetes project runs a community-owned image registry called `registry.k8s.io` to host it's container images. On the 3rd of April 2023, the old registry `k8s.gcr.io` will be frozen and no further images for Kubernetes and related subprojects will be pushed to the old registry.

This registry `registry.k8s.io` replaced the old one and has been generally available (GA) for several months. We have published a [blog post](/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/) about its benefits to community and the Kubernetes project. This post also announced that future versions of Kubernetes will not be available in the old registry. Now that time has come.
```

Author: ack, copied over to the email doc.

What does this change mean for contributors:
- If you are a maintainer of a subproject, you will need to update your manifests and Helm charts to use the new registry.

What does this change mean for end users:

> - 1.27 Kubernetes release will not be published to the old registry.

Reviewer: also patches for older versions of k8s I assume? Let's mention them here

> - Starting in 1.25, the default image registry has been set to `registry.k8s.io`. This value is overridable but will no longer work for any version of Kubernetes released after April as they won’t be present in the old registry.

Reviewer: > This value is overridable but will no longer work for any version of Kubernetes

this sentence is confusing. Overriding it is desirable (to some private registry controlled by user or cloud provider) and will work for all versions. Using the old value will not work.

Please rephrase

Author: I meant overriding the registry with k8s.gcr.io won't be useful after April. I'll rephrase.

> - If you are running Kubernetes in networks where outbound traffic is restricted, you should consider hosting local image registry mirrors.

Reviewer: ```suggestion
- If you want to increase reliability of your cluster and remove dependency on community-owned registry or you are running Kubernetes in networks where external traffic is restricted, you should consider hosting local image registry mirrors. Some cloud vendors may offer hosted solutions for this.
```

Reviewer: let's emphasize that reliability will increase by removing dependency on community owned registry

Author: ack, copied over to the email doc.

Timeline of the Changes:

- `k8s.gcr.io` will be frozen on the 3rd of April 2023
- 1.27 is expected to be released on the 12th of April 2023
- The last 1.23 release on `k8s.gcr.io` will be 1.23.18 (1.23 goes EoL before the freeze)
- The last 1.24 release on `k8s.gcr.io` will be 1.24.12
- The last 1.25 release on `k8s.gcr.io` will be 1.25.8
- The last 1.26 release on `k8s.gcr.io` will be 1.26.3

> 

Reviewer: Can you add a section either here or in documents that will offer a simple script to iterate thru images of all Pods and warn if any of them are using old registry?

- Patch releases for 1.24, 1.25 and 1.26 will no longer be published to the old registry from April. Please read the timelines below for details of the final patch releases in the old registry.
- Starting in 1.25, the default image registry has been set to `registry.k8s.io`. This value is overridable in `kubeadm` and `kubelet` but setting it to `k8s.gcr.io` will fail for new releases after April as they won’t be present in the old registry.
- If you want to increase reliability of your cluster and remove dependency on community-owned registry or you are running Kubernetes in networks where external traffic is restricted, you should consider hosting local image registry mirrors. Some cloud vendors may offer hosted solutions for this.

## Timeline of the Changes:

- `k8s.gcr.io` will be frozen on the 3rd of April 2023
- 1.27 is expected to be released on the 12th of April 2023
- The last 1.23 release on `k8s.gcr.io` will be 1.23.18 (1.23 goes EoL before the freeze)
- The last 1.24 release on `k8s.gcr.io` will be 1.24.12
- The last 1.25 release on `k8s.gcr.io` will be 1.25.8
- The last 1.26 release on `k8s.gcr.io` will be 1.26.3

> You can run this command to list the images used by pods:

Reviewer: ```suggestion

## What's next

Please make sure your cluster does not have dependencies on old image registry. For example,  you can run this command to list the images used by pods:
```

```shell
kubectl get pods --all-namespaces -o jsonpath="{.items[*].spec.containers[*].image}" |\
tr -s '[[:space:]]' '\n' |\
sort |\
uniq -c
```

> 

Reviewer: ```suggestion

There may be other dependencies to old image registry. Make sure to review any potential dependencies to keep your cluster healthy and up to date.
```

---
layout: blog
title: "k8s.gcr.io image registry will be frozen from the 3rd of April 2023"

> date: 2023-02-06

Reviewer 2: ```suggestion
date: 2023-02-06T17:00:00
```

What does this change mean for contributors:
- If you are a maintainer of a subproject, you will need to update your manifests and Helm charts to use the new registry.

What does this change mean for end users:
- 1.27 Kubernetes release will not be published to the old registry.
- Patch releases for 1.24, 1.25, and 1.26 will no longer be published to the old registry from April. Please read the timelines below for details of the final patch releases in the old registry.
- Starting in 1.25, the default image registry has been set to `registry.k8s.io`. This value is overridable in `kubeadm` and `kubelet` but setting it to `k8s.gcr.io` will fail for new releases after April as they won’t be present in the old registry.
- If you want to increase the reliability of your cluster and remove dependency on the community-owned registry or you are running Kubernetes in networks where external traffic is restricted, you should consider hosting local image registry mirrors. Some cloud vendors may offer hosted solutions for this.

> ## Timeline of the Changes:

Reviewer 2: ```suggestion
## Timeline of the changes
```

- `k8s.gcr.io` will be frozen on the 3rd of April 2023
- 1.27 is expected to be released on the 12th of April 2023

> - The last 1.23 release on `k8s.gcr.io` will be 1.23.18 (1.23 goes EoL before the freeze)

Reviewer 2: ```suggestion
- The last 1.23 release on `k8s.gcr.io` will be 1.23.18 (1.23 goes end-of-life before the freeze)
```

Author: applied all the suggestions
